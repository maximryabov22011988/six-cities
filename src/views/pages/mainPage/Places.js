import React from 'react';
import PropTypes from 'prop-types';

import PlaceCardList from './PlaceCardList';
import Select from './Select';

const propTypes = {
  searchResultText: PropTypes.string.isRequired,
  offers: PropTypes.arrayOf(
    PropTypes.shape({
      bedrooms: PropTypes.number,
      city: PropTypes.shape({
        name: PropTypes.string,
        location: PropTypes.object,
      }),
      description: PropTypes.string,
      goods: PropTypes.arrayOf(PropTypes.string),
      host: PropTypes.shape({
        avatar_url: PropTypes.string,
        id: PropTypes.number,
        is_pro: PropTypes.bool,
        name: PropTypes.string,
      }),
      id: PropTypes.number,
      images: PropTypes.arrayOf(PropTypes.string),
      is_favorite: PropTypes.bool,
      is_premium: PropTypes.bool,
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number,
      }),
      max_adults: PropTypes.number,
      preview_image: PropTypes.string,
      price: PropTypes.number,
      rating: PropTypes.number,
      title: PropTypes.string,
      type: PropTypes.string,
    }),
  ).isRequired,
};

const filterOptions = [
  { id: 1, name: 'Popular' },
  { id: 2, name: 'Price: low to high' },
  { id: 3, name: 'Price: high to low' },
  { id: 4, name: 'Top rated first' },
];

const defaultOptionId = 1;

function Places({ searchResultText, offers }) {
  return (
    <section className="cities__places places">
      <h2 className="visually-hidden">Places</h2>
      <b className="places__found">{searchResultText}</b>
      <Select
        caption="Sort by"
        options={filterOptions}
        defaultOption={defaultOptionId}
      />
      <PlaceCardList offers={offers} />
    </section>
  );
}

Places.propTypes = propTypes;

export default Places;
