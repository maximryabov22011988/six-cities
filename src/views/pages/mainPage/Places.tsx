import * as React from 'react';

import PlaceCardList from '../../components/PlaceCardList';
import Select from './places/Select';

import { SORTING_OPTIONS } from '../../constants/options';

import { Offer, onAddToFavorities, onRemoveFromFavorities } from '../../interfaces';

interface Props {
  offers: Array<Offer>,
  searchResultText: string,
  onActiveOfferClick?: ((id: number) => (event: React.MouseEvent<HTMLAnchorElement>) => void) | undefined,
  onAddToFavorities: onAddToFavorities,
  onChangeSorting: (sorting: string) => void,
  onRemoveFromFavorities: onRemoveFromFavorities,
}

const defaultOptionId = 1; // Popular

function Places({
  searchResultText,
  offers,
  onActiveOfferClick,
  onAddToFavorities,
  onChangeSorting,
  onRemoveFromFavorities,
}: Props) {
  return (
    <section className="cities__places places">
      <h2 className="visually-hidden">Places</h2>
      <b className="places__found">{searchResultText}</b>

      <Select
        caption="Sort by"
        defaultOption={defaultOptionId}
        options={SORTING_OPTIONS}
        onChangeSorting={onChangeSorting}
      />

      <PlaceCardList
        offers={offers}
        parentClassName="cities"
        onActiveOfferClick={onActiveOfferClick}
        onAddToFavorities={onAddToFavorities}
        onRemoveFromFavorities={onRemoveFromFavorities}
      />
    </section>
  );
}

export default Places;
