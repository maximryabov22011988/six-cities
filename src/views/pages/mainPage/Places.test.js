import React from 'react';
import renderer from 'react-test-renderer';

import Places from './Places';

const offersMock = [
  {
    id: 'card-5',
    title: 'Wood and stone place',
    image: 'img/room.jpg',
    price: 80,
    type: 'Private room',
    rating: 4,
    isPremium: false,
    isBookmark: true
  }
];

it('Places correctly render', () => {
  const placeCard = renderer
    .create(
      <Places
        searchResultText="1 places to stay in Paris"
        offers={offersMock}
      />
    )
    .toJSON();

  expect(placeCard).toMatchSnapshot();
});
