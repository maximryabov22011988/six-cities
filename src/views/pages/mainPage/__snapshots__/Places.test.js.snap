// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places correctly render 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    1 places to stay in Paris
  </b>
  <div
    className="places__sorting"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      onFocus={[Function]}
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
  </div>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "background": "#e1e1e1",
                "borderRadius": "4px",
                "display": "flex",
                "height": "200px",
                "justifyContent": "center",
                "width": "260px",
              }
            }
          >
            <img
              alt="Place image"
              className="place-card__image"
              height={0}
              onError={[Function]}
              onLoad={[Function]}
              src="img/apartment-01.jpg"
              width={0}
            />
            <span>
              Loading ...
            </span>
          </div>
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "92.00%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
            onClick={[Function]}
          >
            The Pondhouse - A Magical Place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          house
        </p>
      </div>
    </article>
  </div>
</section>
`;
